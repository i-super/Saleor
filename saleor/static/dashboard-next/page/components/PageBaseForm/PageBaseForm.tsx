import Button from "material-ui/Button";
import { CardContent } from "material-ui/Card";
import Grid from "material-ui/Grid";
import { withStyles, WithStyles } from "material-ui/styles";
import TextField from "material-ui/TextField";
import Toolbar from "material-ui/Toolbar";
import Typography from "material-ui/Typography";
import * as React from "react";

import ControlledCheckbox from "../../../components/ControlledCheckbox";
import FormSpacer from "../../../components/FormSpacer";
import RichTextEditor from "../../../components/RichTextEditor";
import i18n from "../../../i18n";

interface PageBaseFormComponentProps {
  availableOn: string;
  content: string;
  created?: string;
  errors?: Array<{ field: string; message: string }>;
  isVisible: boolean;
  slug: string;
  title: string;
  onChange?(event: React.ChangeEvent<any>);
}

const decorate = withStyles(theme => ({
  addHelperTextPadding: {
    [theme.breakpoints.up("md")]: {
      paddingBottom: theme.spacing.unit * 2.5
    }
  },
  cardActions: {
    flexDirection: "row-reverse" as "row-reverse"
  }
}));

export const PageBaseForm = decorate<PageBaseFormComponentProps>(
  ({
    classes,
    created,
    errors,
    title,
    content,
    slug,
    availableOn,
    isVisible,
    onChange
  }) => {
    const errorList: { [key: string]: string } = errors
      ? errors.reduce((acc, curr) => {
          acc[curr.field] = curr.message;
          return acc;
        }, {})
      : {};
    return (
      <CardContent>
        <Grid container spacing={16}>
          <Grid item xs={12} md={9}>
            <TextField
              autoFocus
              name="title"
              label={i18n.t("Title", { context: "object" })}
              value={title}
              onChange={onChange}
              className={
                (errors ? !errorList.title : true)
                  ? classes.addHelperTextPadding
                  : ""
              }
              error={!!(errorList && errorList.title)}
              helperText={errorList && errorList.title ? errorList.title : ""}
              fullWidth
            />
            <FormSpacer />
            <RichTextEditor
              value={content}
              name="content"
              label={i18n.t("Content", { context: "object" })}
              helperText={
                errorList && errorList.content
                  ? errorList.content
                  : i18n.t("Select text to enable text-formatting tools.", {
                      context: "object"
                    })
              }
              onChange={onChange}
              error={!!(errorList && errorList.content)}
              fullWidth
            />
          </Grid>
          <Grid item xs={12} md={3}>
            <TextField
              name="slug"
              label={i18n.t("Slug", { context: "object" })}
              value={slug}
              helperText={
                errorList && errorList.slug
                  ? errorList.slug
                  : i18n.t("Slug is being used to create page URL", {
                      context: "object"
                    })
              }
              onChange={onChange}
              error={!!(errorList && errorList.slug)}
              fullWidth
            />
            <FormSpacer />
            <TextField
              name="availableOn"
              label={i18n.t("Available on", { context: "label" })}
              type="date"
              value={availableOn}
              onChange={onChange}
              InputLabelProps={{
                shrink: true
              }}
              helperText={
                errorList && errorList.availableOn ? errorList.availableOn : ""
              }
              error={!!(errorList && errorList.availableOn)}
            />
            <FormSpacer />
            {created && (
              <Typography variant="body1">Created at: {created}</Typography>
            )}
            <FormSpacer />
            <ControlledCheckbox
              checked={isVisible}
              label={i18n.t("Published", { context: "label" })}
              name="isVisible"
              onChange={onChange}
            />
          </Grid>
        </Grid>
      </CardContent>
    );
  }
);

export default PageBaseForm;
