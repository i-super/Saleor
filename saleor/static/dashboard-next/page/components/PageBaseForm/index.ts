export { default } from "./PageBaseForm";
export * from "./PageBaseForm";
