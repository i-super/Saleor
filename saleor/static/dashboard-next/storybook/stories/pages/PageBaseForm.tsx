import { storiesOf } from "@storybook/react";
import { DialogContentText } from "material-ui/Dialog";
import * as React from "react";

import PageBaseForm from "../../../page/components/PageBaseForm";

const page = {
  availableOn: "2018-03-20T16:39:08.850105+00:00",
  content:
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin bibendum et justo sit amet viverra. Maecenas euismod auctor nisi et tincidunt. Maecenas ante urna, aliquet at odio sed, gravida efficitur purus. Morbi ut dapibus ante. Nulla eu neque sit amet odio tincidunt porttitor eu quis purus. Aliquam at diam sit amet turpis varius semper vitae non lectus. Vivamus lectus ligula, molestie eu augue eu, cursus tincidunt sem. Nam ornare egestas tincidunt.",
  isVisible: true,
  slug: "lorem-ipsum-dolor",
  title: "Lorem ipsum"
};

storiesOf("Pages / PageBaseForm", module)
  .add("with initial data", () => <PageBaseForm {...page} />)
  .add("with 'created' property", () => {
    const created = "2018-03-20T16:39:08.850105+00:00";
    return <PageBaseForm {...page} created={created} />;
  })
  .add("with 'errors' property", () => {
    const errors = [
      {
        field: "slug",
        message: "To pole jest wymagane."
      },
      {
        field: "title",
        message: "To pole jest wymagane."
      },
      {
        field: "content",
        message: "To pole jest wymagane."
      }
    ];
    return <PageBaseForm {...page} errors={errors} />;
  });
