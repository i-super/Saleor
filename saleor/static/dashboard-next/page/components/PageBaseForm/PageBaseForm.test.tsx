import * as React from "react";
import * as renderer from "react-test-renderer";

import PageBaseForm from "./";

const page = {
  availableOn: "2018-03-20T16:39:08.850105+00:00",
  content:
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin bibendum et justo sit amet viverra. Maecenas euismod auctor nisi et tincidunt. Maecenas ante urna, aliquet at odio sed, gravida efficitur purus. Morbi ut dapibus ante. Nulla eu neque sit amet odio tincidunt porttitor eu quis purus. Aliquam at diam sit amet turpis varius semper vitae non lectus. Vivamus lectus ligula, molestie eu augue eu, cursus tincidunt sem. Nam ornare egestas tincidunt.",
  isVisible: true,
  slug: "lorem-ipsum-dolor",
  title: "Lorem ipsum"
};

describe("<PageBaseForm />", () => {
  it("renders without 'created' property", () => {
    const component = renderer.create(<PageBaseForm {...page} />);
    expect(component).toMatchSnapshot();
  });
  it("renders with 'created' property", () => {
    const created = "2018-03-20T16:39:08.850105+00:00";
    const component = renderer.create(
      <PageBaseForm {...page} created={created} />
    );
    expect(component).toMatchSnapshot();
  });
  it("renders errors", () => {
    const errors = [
      {
        field: "slug",
        message: "To pole jest wymagane."
      },
      {
        field: "title",
        message: "To pole jest wymagane."
      },
      {
        field: "content",
        message: "To pole jest wymagane."
      }
    ];
    const component = renderer.create(
      <PageBaseForm {...page} errors={errors} />
    );
    expect(component).toMatchSnapshot();
  });
});
