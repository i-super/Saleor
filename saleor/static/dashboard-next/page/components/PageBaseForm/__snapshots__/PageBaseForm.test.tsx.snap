// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageBaseForm /> renders errors 1`] = `
<div
  className="MuiCardContent-root-3"
>
  <div
    className="MuiGrid-container-4 MuiGrid-spacing-xs-16-26"
  >
    <div
      className="MuiGrid-item-5 MuiGrid-grid-xs-12-41 MuiGrid-grid-md-9-64"
    >
      <div
        aria-describedby={undefined}
        className="MuiFormControl-root-94 MuiFormControl-fullWidth-97"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-103 MuiFormLabel-error-106 MuiInputLabel-root-98 MuiInputLabel-formControl-99 MuiInputLabel-animated-102 MuiInputLabel-shrink-101"
          data-shrink={true}
          htmlFor={undefined}
        >
          Title
        </label>
        <div
          className="MuiInput-root-108 MuiInput-error-113 MuiInput-fullWidth-115 MuiInput-formControl-109 MuiInput-underline-112"
        >
          <input
            aria-invalid={true}
            aria-required={false}
            autoComplete={undefined}
            autoFocus={true}
            className="MuiInput-input-116"
            defaultValue={undefined}
            disabled={false}
            id={undefined}
            name="title"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={undefined}
            rows={undefined}
            type="text"
            value="Lorem ipsum"
          />
        </div>
        <p
          className="MuiFormHelperText-root-125 MuiFormHelperText-error-126"
          id={undefined}
        >
          To pole jest wymagane.
        </p>
      </div>
      <div
        className="Component-spacer-121"
      />
      <div
        className="MuiFormControl-root-94"
        onBlur={[Function]}
        onFocus={[Function]}
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <label
          className="MuiFormLabel-root-103 MuiFormLabel-error-106 MuiInputLabel-root-98 MuiInputLabel-formControl-99 MuiInputLabel-animated-102 MuiInputLabel-shrink-101"
          data-shrink={true}
        >
          Content
        </label>
        <div
          className="MuiInput-root-108 MuiInput-error-113 MuiInput-fullWidth-115 MuiInput-formControl-109 MuiInput-multiline-114 MuiInput-underline-112"
        >
          <div
            className="Textarea-root-122"
            style={
              Object {
                "height": 19,
              }
            }
          >
            <textarea
              aria-hidden="true"
              className="Textarea-shadow-124 Textarea-textarea-123"
              readOnly={true}
              rows="1"
              tabIndex={-1}
              value=""
            />
            <textarea
              aria-hidden="true"
              className="Textarea-shadow-124 Textarea-textarea-123"
              defaultValue={undefined}
              readOnly={true}
              rows={1}
              tabIndex={-1}
              value="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin bibendum et justo sit amet viverra. Maecenas euismod auctor nisi et tincidunt. Maecenas ante urna, aliquet at odio sed, gravida efficitur purus. Morbi ut dapibus ante. Nulla eu neque sit amet odio tincidunt porttitor eu quis purus. Aliquam at diam sit amet turpis varius semper vitae non lectus. Vivamus lectus ligula, molestie eu augue eu, cursus tincidunt sem. Nam ornare egestas tincidunt."
            />
            <textarea
              aria-invalid={true}
              aria-required={false}
              autoComplete={undefined}
              autoFocus={undefined}
              className="Textarea-textarea-123 MuiInput-input-116 MuiInput-inputMultiline-118"
              defaultValue={undefined}
              disabled={false}
              id={undefined}
              name="content"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={undefined}
              onKeyUp={undefined}
              placeholder={undefined}
              readOnly={undefined}
              required={undefined}
              rows={1}
              type="text"
              value="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin bibendum et justo sit amet viverra. Maecenas euismod auctor nisi et tincidunt. Maecenas ante urna, aliquet at odio sed, gravida efficitur purus. Morbi ut dapibus ante. Nulla eu neque sit amet odio tincidunt porttitor eu quis purus. Aliquam at diam sit amet turpis varius semper vitae non lectus. Vivamus lectus ligula, molestie eu augue eu, cursus tincidunt sem. Nam ornare egestas tincidunt."
            />
          </div>
        </div>
        <p
          className="MuiFormHelperText-root-125 MuiFormHelperText-error-126"
        >
          To pole jest wymagane.
        </p>
      </div>
    </div>
    <div
      className="MuiGrid-item-5 MuiGrid-grid-xs-12-41 MuiGrid-grid-md-3-58"
    >
      <div
        aria-describedby={undefined}
        className="MuiFormControl-root-94 MuiFormControl-fullWidth-97"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-103 MuiFormLabel-error-106 MuiInputLabel-root-98 MuiInputLabel-formControl-99 MuiInputLabel-animated-102 MuiInputLabel-shrink-101"
          data-shrink={true}
          htmlFor={undefined}
        >
          Slug
        </label>
        <div
          className="MuiInput-root-108 MuiInput-error-113 MuiInput-fullWidth-115 MuiInput-formControl-109 MuiInput-underline-112"
        >
          <input
            aria-invalid={true}
            aria-required={false}
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInput-input-116"
            defaultValue={undefined}
            disabled={false}
            id={undefined}
            name="slug"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={undefined}
            rows={undefined}
            type="text"
            value="lorem-ipsum-dolor"
          />
        </div>
        <p
          className="MuiFormHelperText-root-125 MuiFormHelperText-error-126"
          id={undefined}
        >
          To pole jest wymagane.
        </p>
      </div>
      <div
        className="Component-spacer-121"
      />
      <div
        aria-describedby={undefined}
        className="MuiFormControl-root-94"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-103 MuiInputLabel-root-98 MuiInputLabel-formControl-99 MuiInputLabel-animated-102 MuiInputLabel-shrink-101"
          data-shrink={true}
          htmlFor={undefined}
        >
          Available on
        </label>
        <div
          className="MuiInput-root-108 MuiInput-formControl-109 MuiInput-underline-112"
        >
          <input
            aria-invalid={false}
            aria-required={false}
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInput-input-116 MuiInput-inputType-119"
            defaultValue={undefined}
            disabled={false}
            id={undefined}
            name="availableOn"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={undefined}
            rows={undefined}
            type="date"
            value="2018-03-20T16:39:08.850105+00:00"
          />
        </div>
        
      </div>
      <div
        className="Component-spacer-121"
      />
      <div
        className="Component-spacer-121"
      />
      <label
        className="MuiFormControlLabel-root-129"
      >
        <span
          className="MuiButtonBase-root-147 MuiIconButton-root-141 MuiSwitchBase-root-137 MuiCheckbox-root-132 MuiCheckbox-colorSecondary-136 MuiSwitchBase-checked-138 MuiCheckbox-checked-133"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role={undefined}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label-146"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-149"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
            <input
              checked={true}
              className="MuiSwitchBase-input-140"
              disabled={undefined}
              id={undefined}
              name="isVisible"
              onChange={[Function]}
              tabIndex={undefined}
              type="checkbox"
              value={undefined}
            />
          </span>
          <span
            className="MuiTouchRipple-root-155"
          />
        </span>
        <span
          className="MuiTypography-root-162 MuiTypography-body1-171 MuiFormControlLabel-label-131"
        >
          Published
        </span>
      </label>
    </div>
  </div>
</div>
`;

exports[`<PageBaseForm /> renders with 'created' property 1`] = `
<div
  className="MuiCardContent-root-3"
>
  <div
    className="MuiGrid-container-4 MuiGrid-spacing-xs-16-26"
  >
    <div
      className="MuiGrid-item-5 MuiGrid-grid-xs-12-41 MuiGrid-grid-md-9-64"
    >
      <div
        aria-describedby={undefined}
        className="MuiFormControl-root-94 MuiFormControl-fullWidth-97 Component-addHelperTextPadding-1"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-103 MuiInputLabel-root-98 MuiInputLabel-formControl-99 MuiInputLabel-animated-102 MuiInputLabel-shrink-101"
          data-shrink={true}
          htmlFor={undefined}
        >
          Title
        </label>
        <div
          className="MuiInput-root-108 MuiInput-fullWidth-115 MuiInput-formControl-109 MuiInput-underline-112"
        >
          <input
            aria-invalid={false}
            aria-required={false}
            autoComplete={undefined}
            autoFocus={true}
            className="MuiInput-input-116"
            defaultValue={undefined}
            disabled={false}
            id={undefined}
            name="title"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={undefined}
            rows={undefined}
            type="text"
            value="Lorem ipsum"
          />
        </div>
        
      </div>
      <div
        className="Component-spacer-121"
      />
      <div
        className="MuiFormControl-root-94"
        onBlur={[Function]}
        onFocus={[Function]}
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <label
          className="MuiFormLabel-root-103 MuiInputLabel-root-98 MuiInputLabel-formControl-99 MuiInputLabel-animated-102 MuiInputLabel-shrink-101"
          data-shrink={true}
        >
          Content
        </label>
        <div
          className="MuiInput-root-108 MuiInput-fullWidth-115 MuiInput-formControl-109 MuiInput-multiline-114 MuiInput-underline-112"
        >
          <div
            className="Textarea-root-122"
            style={
              Object {
                "height": 19,
              }
            }
          >
            <textarea
              aria-hidden="true"
              className="Textarea-shadow-124 Textarea-textarea-123"
              readOnly={true}
              rows="1"
              tabIndex={-1}
              value=""
            />
            <textarea
              aria-hidden="true"
              className="Textarea-shadow-124 Textarea-textarea-123"
              defaultValue={undefined}
              readOnly={true}
              rows={1}
              tabIndex={-1}
              value="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin bibendum et justo sit amet viverra. Maecenas euismod auctor nisi et tincidunt. Maecenas ante urna, aliquet at odio sed, gravida efficitur purus. Morbi ut dapibus ante. Nulla eu neque sit amet odio tincidunt porttitor eu quis purus. Aliquam at diam sit amet turpis varius semper vitae non lectus. Vivamus lectus ligula, molestie eu augue eu, cursus tincidunt sem. Nam ornare egestas tincidunt."
            />
            <textarea
              aria-invalid={false}
              aria-required={false}
              autoComplete={undefined}
              autoFocus={undefined}
              className="Textarea-textarea-123 MuiInput-input-116 MuiInput-inputMultiline-118"
              defaultValue={undefined}
              disabled={false}
              id={undefined}
              name="content"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={undefined}
              onKeyUp={undefined}
              placeholder={undefined}
              readOnly={undefined}
              required={undefined}
              rows={1}
              type="text"
              value="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin bibendum et justo sit amet viverra. Maecenas euismod auctor nisi et tincidunt. Maecenas ante urna, aliquet at odio sed, gravida efficitur purus. Morbi ut dapibus ante. Nulla eu neque sit amet odio tincidunt porttitor eu quis purus. Aliquam at diam sit amet turpis varius semper vitae non lectus. Vivamus lectus ligula, molestie eu augue eu, cursus tincidunt sem. Nam ornare egestas tincidunt."
            />
          </div>
        </div>
        <p
          className="MuiFormHelperText-root-125"
        >
          Select text to enable text-formatting tools.
        </p>
      </div>
    </div>
    <div
      className="MuiGrid-item-5 MuiGrid-grid-xs-12-41 MuiGrid-grid-md-3-58"
    >
      <div
        aria-describedby={undefined}
        className="MuiFormControl-root-94 MuiFormControl-fullWidth-97"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-103 MuiInputLabel-root-98 MuiInputLabel-formControl-99 MuiInputLabel-animated-102 MuiInputLabel-shrink-101"
          data-shrink={true}
          htmlFor={undefined}
        >
          Slug
        </label>
        <div
          className="MuiInput-root-108 MuiInput-fullWidth-115 MuiInput-formControl-109 MuiInput-underline-112"
        >
          <input
            aria-invalid={false}
            aria-required={false}
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInput-input-116"
            defaultValue={undefined}
            disabled={false}
            id={undefined}
            name="slug"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={undefined}
            rows={undefined}
            type="text"
            value="lorem-ipsum-dolor"
          />
        </div>
        <p
          className="MuiFormHelperText-root-125"
          id={undefined}
        >
          Slug is being used to create page URL
        </p>
      </div>
      <div
        className="Component-spacer-121"
      />
      <div
        aria-describedby={undefined}
        className="MuiFormControl-root-94"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-103 MuiInputLabel-root-98 MuiInputLabel-formControl-99 MuiInputLabel-animated-102 MuiInputLabel-shrink-101"
          data-shrink={true}
          htmlFor={undefined}
        >
          Available on
        </label>
        <div
          className="MuiInput-root-108 MuiInput-formControl-109 MuiInput-underline-112"
        >
          <input
            aria-invalid={false}
            aria-required={false}
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInput-input-116 MuiInput-inputType-119"
            defaultValue={undefined}
            disabled={false}
            id={undefined}
            name="availableOn"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={undefined}
            rows={undefined}
            type="date"
            value="2018-03-20T16:39:08.850105+00:00"
          />
        </div>
        
      </div>
      <div
        className="Component-spacer-121"
      />
      <p
        className="MuiTypography-root-162 MuiTypography-body1-171"
      >
        Created at: 
        2018-03-20T16:39:08.850105+00:00
      </p>
      <div
        className="Component-spacer-121"
      />
      <label
        className="MuiFormControlLabel-root-129"
      >
        <span
          className="MuiButtonBase-root-147 MuiIconButton-root-141 MuiSwitchBase-root-137 MuiCheckbox-root-132 MuiCheckbox-colorSecondary-136 MuiSwitchBase-checked-138 MuiCheckbox-checked-133"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role={undefined}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label-146"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-149"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
            <input
              checked={true}
              className="MuiSwitchBase-input-140"
              disabled={undefined}
              id={undefined}
              name="isVisible"
              onChange={[Function]}
              tabIndex={undefined}
              type="checkbox"
              value={undefined}
            />
          </span>
          <span
            className="MuiTouchRipple-root-155"
          />
        </span>
        <span
          className="MuiTypography-root-162 MuiTypography-body1-171 MuiFormControlLabel-label-131"
        >
          Published
        </span>
      </label>
    </div>
  </div>
</div>
`;

exports[`<PageBaseForm /> renders without 'created' property 1`] = `
<div
  className="MuiCardContent-root-3"
>
  <div
    className="MuiGrid-container-4 MuiGrid-spacing-xs-16-26"
  >
    <div
      className="MuiGrid-item-5 MuiGrid-grid-xs-12-41 MuiGrid-grid-md-9-64"
    >
      <div
        aria-describedby={undefined}
        className="MuiFormControl-root-94 MuiFormControl-fullWidth-97 Component-addHelperTextPadding-1"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-103 MuiInputLabel-root-98 MuiInputLabel-formControl-99 MuiInputLabel-animated-102 MuiInputLabel-shrink-101"
          data-shrink={true}
          htmlFor={undefined}
        >
          Title
        </label>
        <div
          className="MuiInput-root-108 MuiInput-fullWidth-115 MuiInput-formControl-109 MuiInput-underline-112"
        >
          <input
            aria-invalid={false}
            aria-required={false}
            autoComplete={undefined}
            autoFocus={true}
            className="MuiInput-input-116"
            defaultValue={undefined}
            disabled={false}
            id={undefined}
            name="title"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={undefined}
            rows={undefined}
            type="text"
            value="Lorem ipsum"
          />
        </div>
        
      </div>
      <div
        className="Component-spacer-121"
      />
      <div
        className="MuiFormControl-root-94"
        onBlur={[Function]}
        onFocus={[Function]}
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <label
          className="MuiFormLabel-root-103 MuiInputLabel-root-98 MuiInputLabel-formControl-99 MuiInputLabel-animated-102 MuiInputLabel-shrink-101"
          data-shrink={true}
        >
          Content
        </label>
        <div
          className="MuiInput-root-108 MuiInput-fullWidth-115 MuiInput-formControl-109 MuiInput-multiline-114 MuiInput-underline-112"
        >
          <div
            className="Textarea-root-122"
            style={
              Object {
                "height": 19,
              }
            }
          >
            <textarea
              aria-hidden="true"
              className="Textarea-shadow-124 Textarea-textarea-123"
              readOnly={true}
              rows="1"
              tabIndex={-1}
              value=""
            />
            <textarea
              aria-hidden="true"
              className="Textarea-shadow-124 Textarea-textarea-123"
              defaultValue={undefined}
              readOnly={true}
              rows={1}
              tabIndex={-1}
              value="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin bibendum et justo sit amet viverra. Maecenas euismod auctor nisi et tincidunt. Maecenas ante urna, aliquet at odio sed, gravida efficitur purus. Morbi ut dapibus ante. Nulla eu neque sit amet odio tincidunt porttitor eu quis purus. Aliquam at diam sit amet turpis varius semper vitae non lectus. Vivamus lectus ligula, molestie eu augue eu, cursus tincidunt sem. Nam ornare egestas tincidunt."
            />
            <textarea
              aria-invalid={false}
              aria-required={false}
              autoComplete={undefined}
              autoFocus={undefined}
              className="Textarea-textarea-123 MuiInput-input-116 MuiInput-inputMultiline-118"
              defaultValue={undefined}
              disabled={false}
              id={undefined}
              name="content"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={undefined}
              onKeyUp={undefined}
              placeholder={undefined}
              readOnly={undefined}
              required={undefined}
              rows={1}
              type="text"
              value="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin bibendum et justo sit amet viverra. Maecenas euismod auctor nisi et tincidunt. Maecenas ante urna, aliquet at odio sed, gravida efficitur purus. Morbi ut dapibus ante. Nulla eu neque sit amet odio tincidunt porttitor eu quis purus. Aliquam at diam sit amet turpis varius semper vitae non lectus. Vivamus lectus ligula, molestie eu augue eu, cursus tincidunt sem. Nam ornare egestas tincidunt."
            />
          </div>
        </div>
        <p
          className="MuiFormHelperText-root-125"
        >
          Select text to enable text-formatting tools.
        </p>
      </div>
    </div>
    <div
      className="MuiGrid-item-5 MuiGrid-grid-xs-12-41 MuiGrid-grid-md-3-58"
    >
      <div
        aria-describedby={undefined}
        className="MuiFormControl-root-94 MuiFormControl-fullWidth-97"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-103 MuiInputLabel-root-98 MuiInputLabel-formControl-99 MuiInputLabel-animated-102 MuiInputLabel-shrink-101"
          data-shrink={true}
          htmlFor={undefined}
        >
          Slug
        </label>
        <div
          className="MuiInput-root-108 MuiInput-fullWidth-115 MuiInput-formControl-109 MuiInput-underline-112"
        >
          <input
            aria-invalid={false}
            aria-required={false}
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInput-input-116"
            defaultValue={undefined}
            disabled={false}
            id={undefined}
            name="slug"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={undefined}
            rows={undefined}
            type="text"
            value="lorem-ipsum-dolor"
          />
        </div>
        <p
          className="MuiFormHelperText-root-125"
          id={undefined}
        >
          Slug is being used to create page URL
        </p>
      </div>
      <div
        className="Component-spacer-121"
      />
      <div
        aria-describedby={undefined}
        className="MuiFormControl-root-94"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-103 MuiInputLabel-root-98 MuiInputLabel-formControl-99 MuiInputLabel-animated-102 MuiInputLabel-shrink-101"
          data-shrink={true}
          htmlFor={undefined}
        >
          Available on
        </label>
        <div
          className="MuiInput-root-108 MuiInput-formControl-109 MuiInput-underline-112"
        >
          <input
            aria-invalid={false}
            aria-required={false}
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInput-input-116 MuiInput-inputType-119"
            defaultValue={undefined}
            disabled={false}
            id={undefined}
            name="availableOn"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={undefined}
            rows={undefined}
            type="date"
            value="2018-03-20T16:39:08.850105+00:00"
          />
        </div>
        
      </div>
      <div
        className="Component-spacer-121"
      />
      <div
        className="Component-spacer-121"
      />
      <label
        className="MuiFormControlLabel-root-129"
      >
        <span
          className="MuiButtonBase-root-147 MuiIconButton-root-141 MuiSwitchBase-root-137 MuiCheckbox-root-132 MuiCheckbox-colorSecondary-136 MuiSwitchBase-checked-138 MuiCheckbox-checked-133"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role={undefined}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label-146"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-149"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
            <input
              checked={true}
              className="MuiSwitchBase-input-140"
              disabled={undefined}
              id={undefined}
              name="isVisible"
              onChange={[Function]}
              tabIndex={undefined}
              type="checkbox"
              value={undefined}
            />
          </span>
          <span
            className="MuiTouchRipple-root-155"
          />
        </span>
        <span
          className="MuiTypography-root-162 MuiTypography-body1-171 MuiFormControlLabel-label-131"
        >
          Published
        </span>
      </label>
    </div>
  </div>
</div>
`;
